app/frontend/src/features/auth/login/UsernameForm.tsx

```diff
@@ -57,7 +57,7 @@ export default function UsernameForm() {
           break;
       }
     } catch (e) {
-      dispatch(authError(e.message));
+      authActions.authError(e.message);
     }
     setLoading(false);
   });
```

Comment:
The setState on unmounted error is still there, and the safeState actually doesn't do anything. After some more digging... it looks like the **complete** conditional render below is the culprit so I suspect it's complaining about setState as the rendered component gets unmounted when they switch.

Inlining the `TextBody` stuff with `sent && <TextBody>...` so it's all in one return statement fixed it, but then that means you'll have to do another check to make sure the form only renders when it's `!sent`...

Reply:
Fixed but I must admit I don't understand why the conditional render was causing this...

Reply:
Yeah it's a bit odd, since the component rendering it still exists. It's as if that unmounting means the `<TextBody>` gets a new copy of state all over again? 🤷